src/Algebra/Action/Bundles.agda

```diff
@@ -0,0 +1,141 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Setoid Actions and Monoid Actions
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Action.Bundles where
+
+open import Algebra.Action.Structures.Raw using (IsRawLeftAction; IsRawRightAction)
+open import Algebra.Bundles using (Monoid)
+
+open import Data.List.Base using ([]; _++_)
+import Data.List.Relation.Binary.Equality.Setoid as ≋
+open import Data.Product.Base using (curry; uncurry)
+open import Data.Product.Relation.Binary.Pointwise.NonDependent using (_×ₛ_)
+
+open import Function.Bundles using (Func)
+
+open import Level using (Level; _⊔_)
+
+open import Relation.Binary.Bundles using (Setoid)
+
+private
+  variable
+    a c r ℓ : Level
+
+
+------------------------------------------------------------------------
+-- Basic definition: a Setoid action yields underlying raw action
+
+module SetoidAction (M : Setoid c ℓ) (A : Setoid a r) where
+
+  private
+
+    open module M = Setoid M using ()
+    open module A = Setoid A using (_≈_)
+
+  record Left : Set (a ⊔ r ⊔ c ⊔ ℓ) where
+
+    field
+      act : Func (M ×ₛ A) A
+
+    isRawLeftAction : IsRawLeftAction M._≈_ _≈_
+    isRawLeftAction = record { _ᴹ∙ᴬ_ = curry to ; ∙-cong = curry cong }
+      where open Func act
+
+  record Right : Set (a ⊔ r ⊔ c ⊔ ℓ) where
+
+    field
+      act : Func (A ×ₛ M) A
+
+    isRawRightAction : IsRawRightAction M._≈_ _≈_
+    isRawRightAction = record { _ᴬ∙ᴹ_ = curry to ; ∙-cong = curry cong }
+      where open Func act
+
+
+------------------------------------------------------------------------
+-- A Setoid action yields an iterated List action
+
+module _ {M : Setoid c ℓ} {A : Setoid a r} where
+
+  open SetoidAction
+
+  open ≋ M using (≋-setoid)
+
+  leftListAction : (leftAction : Left M A) → Left ≋-setoid A
```

Comment:
1. The `List` is the *carrier* of the `≋-setoid` introduced at L66
2. Initially I had had `LeftAction` etc., but now that they fall under module names `SetoidAction` and `MonoidAction` (because `Setoid` and `Monoid` are already in scope... ;-)), I couldn't stand the additional redundancy involved in writing `Action` twice... unqualified that would lead to
```agda
Algebra.Action.Bundles.MonoidAction.LeftAction
```
in which the cue word '`Action`' appears 3 times!

Reply:
See recent commit on naming.

Reply:
I think what confused me was that I could remember 'trivially' that `List` was the carrier of `≋-setoid`. I'm thinking that if a particular setoid has a known, particular carrier, maybe that too should be part of its name?

Reply:
Conceivably, and from a ux perspective, I perhaps even agree with you. But are you suggesting I open the instantiated module *renaming* `≋-setoid` to include `List` somewhere in its name? That's a can of worms to want to open up!?

Reply:
If the naming of `≋-setoid` is external to this PR, then that would be feature creep beyond what it reasonable to ask for. Trying to fix it locally for readability is not worth doing (sad as that may be.)